WifiART/src/userinterface/ArtMvcView.java

```diff
@@ -174,90 +189,355 @@ private void setupOutputFileChooserPanel(JPanel panel) {
+     *
+     * @return whether or not the Z subelement is included.
+     */
     public boolean getZIncluded() {
         return zCheckbox.isSelected();
     }
+
+    /**
+     * Get whether or not the Usage Rules/Policy subelement is to be included in the output
+     *
+     * @return whether or not the Usage Rules/Policy subelement is included.
+     */
     public boolean getUsageIncluded() {
         return usageCheckbox.isSelected();
     }
+
+    /**
+     * Get whether or not the BSSID List subelement is to be included in the output
+     *
+     * @return whether or not the BSSID List subelement is included.
+     */
     public boolean getBssidIncluded() {
         return bssidCheckbox.isSelected();
     }
+
+    /**
+     * Get whether or not the Location Civic subelement is to be included in the output
+     *
+     * @return whether or not the Location Civic subelement is included.
+     */
     public boolean getLcrIncluded() {
         return lcrCheckbox.isSelected();
     }
+
+    /**
+     * Get whether or not the Map Image subelement is to be included in the output
+     *
+     * @return whether or not the Map Image subelement is included.
+     */
     public boolean getMapIncluded() {
         return mapCheckbox.isSelected();
     }
 
+    /**
+     * Get the name of the input file.
+     *
+     * @return the name of the input file
+     */
     public String getInputFileName() {
         return inputFileNameField.getText();
     }
+
+    /**
+     * Get the input file directory.
+     *
+     * @return the input file directory
+     */
     public String getInputDir() {
         return inputDirField.getText();
     }
+
+    /**
+     * Get the name of the output file.
+     *
+     * @return the name of the output file
+     */
     public String getOutputFileName() {
         return outputFileNameField.getText();
     }
+
+    /**
+     * Get the output file directory.
+     *
+     * @return the output file directory
+     */
     public String getOutputDir() {
         return outputDirField.getText();
     }
 
-    public void setLciIncludedListener(ActionListener listener) {
+    /**
+     * Get whether or not the buffer should be displayed in a readable format.
+     *
+     * @return whether or not the buffer should be displayed in a readable format.
+     */
+    public boolean getReadable() {
+        return readableBufferCheckbox.isSelected();
+    }
+
+    /**
+     * Get whether or not the encoding should match the new Android version.
+     *
+     * @return whether or not the encoding should match the new Android version.
+     */
+    public boolean isAndroidVersionNew() {
+        return androidVersionCheckbox.isSelected();
+    }
+
+    /**
+     * Gets which tab is currently selected.
+     *
+     * @return the tab currently selected
+     */
+    public SubelementName getSelectedTab() {
+        int index = tabbedPanel.getSelectedIndex();
+        switch (index) {
+            case 1:
```

Comment:
The index is related to how the tabbedPanel is constructed. If that changes in the future the mapping here will be incorrect. Is there a way to maintain the index <-> sublement mapping somewhere in a central place?

Reply:
I can declare this mapping as a constant. One potential issue is that the order in which tabs are added to the tabbed pane may be switched around without this constant being affected.